docs/core/concepts/results.md

```diff
@@ -94,42 +105,40 @@ b'\x80\x05K\x03.'
 3
 ```
 
-However, depending on your situation and the size of your data, you can configure result handlers to serialize a task's entire `Result`, not just its metadata. For example, Core's base `JSONResultHandler` serializes the entire `Result` object as its return value. If you have a task which returns a small piece of data such as a string, or a short list of numbers, handling that data directly in the returned JSON string will suffice!
+One special result subclass is the `PrefectResult`, which stores the data in a Prefect state database, either Prefect Core's server or Prefect Cloud. You might consider this storage backend since it requires no extra configuration as long as you have tasks which return small pieces of JSON-serializable data. How this actually works is that both the `PrefectResult.value` and `PrefectResult.location` match, so that the `PrefectResult.location` serialized to the state database actually is the value of the result itself. 
 
-To showcase this, below is the same example with a flow configured to checkpoint each task's output using the `JSONResultHandler`. Note that now both `Result.value` and `Result.safe_value` contain references to the actual return value of its task, the integer `3`.
+To showcase this, below is the same example with a flow configured to checkpoint each task's output using the `PrefectResult`. Note that now both `Result.value` and `Result.location` contain references to the actual return value of its task, the integer `3`.
 
 ```python
 >>> state.result[first_result]._result.value                 
 3
 >>> state.result[first_result]._result.safe_value            
-<SafeResult: '3'>
+3
```

Comment:
The `safe_value` access pattern will need to be updated to use `location`.  Note also that the `location` of a PrefectResult is the JSON representation (so in this case it's a string, not an int)

Reply:
Fixed in 891399212da7d55014da260a08f1a3283c8e4c4a